charts/super-agent-deployment/tests/auth_secret_test.yaml

```diff
@@ -0,0 +1,111 @@
+        equal:
+          path: spec.template.spec.containers[0].volumeMounts
+          value:
+            - name: super-agent-config
+              mountPath: /etc/newrelic-super-agent
+              readOnly: true
+            - mountPath: /var/lib/newrelic-super-agent
+              name: var-lib-newrelic-super-agent
+              readOnly: false
+            - name: auth-secret-volume
+              mountPath: /etc/newrelic-super-agent
+              readOnly: true
+      - template: templates/deployment-superagent.yaml
+        equal:
+          path: spec.template.spec.volumes
+          value:
+            - name: super-agent-config
+              configMap:
+                name: local-data-super-agent
+                items:
+                  - key: local_config
+                    path: config.yaml
+            - name: var-lib-newrelic-super-agent
+              emptyDir: {}
+            - name: auth-secret-volume
+              secret:
+                secretName: auth-credentials
+      - template: templates/auth-secret.yaml
+        equal:
+          path: metadata.name
+          value: auth-credentials
+      - template: templates/auth-secret.yaml
+        equal:
+          path: data.auth_key
+          value: dGVzdC1rZXk= # base64 for "test-key"
+
+  - it: no mount and secret is created when authSecret.create is false
+    set:
+      cluster: test
+      authSecret:
+        create: false
+    asserts:
+      - template: templates/deployment-superagent.yaml
+        notContains:
+          path: spec.template.spec.containers[0].volumeMounts
+          value:
+            name: auth-secret-volume
+      - template: templates/deployment-superagent.yaml
+        notContains:
+          path: spec.template.spec.volumes
+          value:
+            name: auth-secret-volume
+
+  - it: when authSecret is empty it defaults to authSecret.create=false
+    set:
+      cluster: test
+      authSecret:
+    asserts:
+      - template: templates/deployment-superagent.yaml
+        notContains:
+          path: spec.template.spec.containers[0].volumeMounts
+          value:
+            name: auth-secret-volume
+      - template: templates/deployment-superagent.yaml
+        notContains:
+          path: spec.template.spec.volumes
+          value:
+            name: auth-secret-volume
+
+  - it: authSecret creation fails when data is not provided
+    set:
+      cluster: test
+      authSecret:
+        create: true
+    asserts:
+      - template: templates/auth-secret.yaml
+        failedTemplate:
+          errorMessage: authSecret.data must be provided when authSecret.create is true
+
+  - it: authSecret creation fails when no auth_key is provided
+    set:
+      cluster: test
+      authSecret:
+        create: true
+        data:
+          auth_key:
+    asserts:
+      - template: templates/auth-secret.yaml
+        failedTemplate:
+          errorMessage: auth_key must be provided when authSecret.create is true
```

Comment:
Should we add a test-case for when `authSecret.create` is not set but there is other fields in `authSecret`?